app/helpers/generic_file_helper.rb

```diff
@@ -21,8 +21,9 @@ def render_download_link text = nil
     link_to (text || "Download"), download_path(@generic_file), { id: "file_download", target: "_new", data: { label: @generic_file.id } }
   end
 
-  def download_path gf
-    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id)
+  # sufia.download path is from ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path
+  def download_path gf, options = {}
+    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id, options)
```

Comment:
1) I recall having casting problems without using GenericFile.find.  In some situations it was returning a SolrDocument (or something like that) which doesn't have the .doi_url attribute.

2) I want the method to return a url in either case.  

3) Maybe I'm assuming too much. My understanding is that my download_path method would return a doi_url if not falsey otherwise do the normal thing for sufia.download_path.